### PR TITLE
kmsv2: ignore cache key expiration on reads

### PR DESCRIPTION
**Pull request by enj**
/kind feature

```release-note
NONE
```

### CODE REVIEW

**Comment by mikedanese** [comment]
Unit test failures are real.

/lgtm

**Review by aramase — COMMENTED** [review]
/lgtm

File: staging/src/k8s.io/apimachinery/pkg/util/cache/expiring_test.go [hunk 1]
```diff
@@ -101,6 +101,17 @@ func TestExpiration(t *testing.T) {
...
 		t.Fatalf("we should have found a key")
 	}
+
+	// Check getting an expired key with and without AllowExpiredGet
+	c.Set("b", "b", time.Second)
+	fc.Step(2 * time.Second)
+	if _, ok := c.Get("b"); ok {
+		t.Fatalf("we should not have found b key")
+	}
+	c.AllowExpiredGet = true
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/util/cache/expiring_test.go:111]: add a Set to be sure this properly clears the data even for the `AllowExpiredGet` case.

Comment (enj) [reply staging/src/k8s.io/apimachinery/pkg/util/cache/expiring_test.go:114]: Done.

**Pull request merged**